src/ast/passes/codegen_llvm.h

```diff
@@ -172,7 +172,9 @@ class CodegenLLVM : public Visitor {
                  FunctionType *func_type);
 
   [[nodiscard]] ScopedExprDeleter accept(Node *node);
-  [[nodiscard]] std::tuple<Value *, ScopedExprDeleter> getMapKey(Map &map);
+  [[nodiscard]] std::tuple<Value *, ScopedExprDeleter> getMapKey(
+      Map &map,
+      Expression *key_expr);
```

Comment:
Could we rather overload `getMapKey` to have both one- and two-parameter variants? It looks like `getMapKey(map, map.map_expr)` is used quite often.